pkgs/tools/misc/taoup/default.nix

```
@@ -0,0 +1,63 @@
+{ lib, stdenv, fetchFromGitHub, ruby, bundlerEnv, bash, ncurses }:
+let
+  gems = bundlerEnv {
+    name = "taoup-env";
+    inherit ruby;
+    gemdir = ./.;
+  };
+
+in
+stdenv.mkDerivation rec {
+  pname = "taoup";
+  version = "1.1.13";
+
+  src = fetchFromGitHub {
+    owner = "globalcitizen";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1qwx88pzkyznzg7z55qksxrw3vh86zz9amr9w78i227ahrm364bx";
+  };
+
+  buildInputs = [ gems ruby ncurses ];
+
+  patches = [
+    # Pre-create a cache within this derivation
+    ./cachedir.patch
+    # Remove the need to test for `tput`, depend on ncurses directly
+    ./tput.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace taoup --subst-var-by ncurses ${ncurses}
+    substituteInPlace taoup-fortune \
+      --subst-var-by out $out \
+      --replace "/bin/bash" "${bash}/bin/bash"
+  '';
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+        mkdir -p $out/{bin,lib/taoup}
+
+        cp taoup $out/lib/taoup
+        cat > $out/bin/taoup <<EOF
+    #!/bin/sh -e
+    exec ${gems}/bin/bundle exec ${ruby}/bin/ruby "$out/lib/taoup/taoup" "\$@"
+    EOF
+        chmod +x $out/bin/taoup
+
+        # Populate the cache created by cachedir.patch above
+        $out/bin/taoup > $out/lib/taoup/cache
+
+        cp taoup-fortune $out/bin
+        chmod +x $out/bin/taoup-fortune
+  '';
+
+  meta = {
+    description = "The Tao of Unix Programming (Ruby-powered ANSI colored fortunes)";
+    homepage = "https://github.com/globalcitizen/taoup";
+    license = lib.licenses.gpl3;
```

```suggestion
    license = lib.licenses.gpl3Plus;
```

---

Let me recheck - there's no "or later" provision in https://github.com/globalcitizen/taoup/blob/master/LICENSE so this might actually be

```suggestion
    license = lib.licenses.gpl3Only;
```

instead.

(https://github.com/globalcitizen/taoup/blob/master/taoup itself doesn't have the recommended notice from the end of the LICENSE file - probably worth an issue to upstream.)

---

Asking in https://github.com/globalcitizen/taoup/issues/21

---

Yep, this ended up `gpl3Only` now.  Thanks for pointing this out!